frontend/src/app/App.tsx

```diff
@@ -136,26 +141,42 @@ const App = () => {
                     <Redirect to={{ ...location, pathname: "/queue" }} />
                   )}
                 />
-                <Route
+                <ProtectedRoute
```

Comment:
This seems like a good encapsulation that's hampered by our current app setup. Without some sort of client side data store that keeps itself refreshed I can't think of any really good way to prevent all this ugly prop drilling. Ideally only `ProtectedRoute` would need to know the permissions. So although we can tweak this by using something like React Context (or continue to expand our use of Redux) I'm not sure it's making things a lot better.